test/Test/Database/LSMTree/Internal/Snapshot/Codec/Golden.hs

```diff
@@ -195,56 +218,128 @@ enumerateTableConfig =
     , (g, merge ) <- [("G0", OneShot), ("G1", Incremental)]
     ]
 
-enumerateSnapLevels :: [(ComponentAnnotation, SnapLevels RunNumber)]
+enumerateSnapLevels :: [(ComponentAnnotation, SnapLevels SnapshotRun)]
 enumerateSnapLevels = fmap (SnapLevels . V.singleton) <$> enumerateSnapLevel
 
 {----------------
 Enumeration of SnapLevel sub-components
 ----------------}
 
-enumerateSnapLevel :: [(ComponentAnnotation, SnapLevel RunNumber)]
+enumerateSnapLevel :: [(ComponentAnnotation, SnapLevel SnapshotRun)]
 enumerateSnapLevel = do
   (a, run) <- enumerateSnapIncomingRun
-  (b, vec) <- enumerateVectorRunNumber
+  (b, vec) <- enumerateVectorRunInfo
   [( fuseAnnotations [ a, b ], SnapLevel run vec)]
 
-enumerateSnapIncomingRun :: [(ComponentAnnotation, SnapIncomingRun RunNumber)]
+enumerateSnapIncomingRun :: [(ComponentAnnotation, SnapIncomingRun SnapshotRun)]
 enumerateSnapIncomingRun =
   let
       inSnaps =
         [ (fuseAnnotations ["R1", a, b],
-           SnapMergingRun policy numRuns mergeDebt nominalCredits sState)
+           SnapMergingRun policy nominalDebt nominalCredits sState)
         | (a, policy ) <- [("P0", LevelTiering), ("P1", LevelLevelling)]
-        , numRuns <- NumRuns <$> [ magicNumber1 ]
-        , mergeDebt      <- MR.MergeDebt    <$> [ magicNumber2 ]
-        , nominalCredits <- NominalCredits <$>  [ magicNumber1 ]
+        , nominalDebt    <- NominalDebt    <$> [ magicNumber2 ]
+        , nominalCredits <- NominalCredits <$> [ magicNumber1 ]
         , (b, sState ) <- enumerateSnapMergingRunState enumerateLevelMergeType
         ]
   in  fold
-      [ [(fuseAnnotations $ "R0" : replicate 4 blank, SnapSingleRun enumerateRunNumbers)]
+      [ [(fuseAnnotations $ "R0" : replicate 4 blank, SnapSingleRun enumerateOpenRunInfo)]
       , inSnaps
       ]
 
 enumerateSnapMergingRunState ::
-     [(ComponentAnnotation, t)] -> [(ComponentAnnotation, SnapMergingRunState t RunNumber)]
+     [(ComponentAnnotation, t)]
+  -> [(ComponentAnnotation, SnapMergingRunState t SnapshotRun)]
 enumerateSnapMergingRunState mTypes =
-  (fuseAnnotations ["C0", blank, blank], SnapCompletedMerge enumerateRunNumbers) :
-    [ (fuseAnnotations ["C1", a, b], SnapOngoingMerge runVec mType)
-    | (a, runVec ) <- enumerateVectorRunNumber
+    [ (fuseAnnotations ["C0", blank, blank],
+       SnapCompletedMerge numRuns mergeDebt enumerateOpenRunInfo)
+    | numRuns   <- NumRuns <$> [ magicNumber1 ]
+    , mergeDebt <- (MR.MergeDebt. MR.MergeCredits) <$> [ magicNumber2 ]
+    ]
+ ++ [ (fuseAnnotations ["C1", a, b],
+       SnapOngoingMerge runParams mergeCredits runVec mType)
+    | let runParams =
+            MR.RunParams {
+              MR.runParamCaching = NoCacheRunData,
+              MR.runParamAlloc   = RunAllocFixed 10,
+              MR.runParamIndex   = Compact
+            }
```

Comment:
Maybe separate this out into a proper enumeration?

Reply:
semi-done, left it as a single value but with a TODO to suggest having a proper enumeration, but avoiding a combinatorial explosion